design/one-pager-pluggable-secret-stores.md

```diff
@@ -0,0 +1,233 @@
+# Pluggable Secret Stores
+
+* Owner: Hasan Türken (@turkenh)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Background
+
+In the [v1.7.0] release, we have introduced support for publishing connection
+details secrets to [External Secret Stores] as an alpha feature. With this
+initial iteration, we have focused on the _in-tree_ implementation of secret
+stores by adding [Hashicorp's Vault] support as our first External Secret Store.
+However, this approach introduces [additional dependencies] to Crossplane core
+and would not scale once the community wants to support more secret stores.
+Even further, it won't be possible to use an in-house secret store with
+Crossplane since adding support for such a store to Crossplane core wouldn't
+make sense. For this purpose, with the initial design, we set it as a [goal] to
+support switching to an out-of-tree (plugin) model but intentionally left it
+[out of scope] to avoid the upfront complexity.
+
+With the gained confidence, we can now iterate with a more scalable option by
+supporting pluggable secret stores.
+
+## Goals
+
+The goal of this document is to propose a pluggable secret store mechanism
+that enables the development and consumption of out-of-tree secret store
+plugins.
+
+This new mechanism should:
+- Allow building/using new secret stores for Crossplane without making any
+changes in Crossplane and/or Provider repositories.
+- Provide a smooth development and consumption experience. 
+- Have secure communication with the plugin process.
```

Comment:
Do we need to support more than one kind of external secret store being deployed?

Reply:
As mentioned somewhere below, 

```
A _Secret Store Plugin_:

- May support one or more Secret Stores.
- ...
```

All `rpc` methods have configReference as their first parameter. So, one secret store plugin may support more than one Secret Store type, e.g. Vault, AWS, GCP, Azure ... all in a single plugin binary, and related implementation could be called depending on a given config. 

Added a new section with the title: **Plugin Binary vs Secret Store Implementation**.